internal/controller/conditions/conditions.go

```diff
@@ -0,0 +1,83 @@
+package conditions
+
+import (
+	"context"
+	"fmt"
+
+	siteconfigv1 "github.com/sakhoury/siteconfig/api/v1alpha1"
+	"github.com/sakhoury/siteconfig/internal/controller/retry"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ConditionType is a string representing the condition's type
+type ConditionType string
+
+// ConditionTypes define the different types of conditions that will be set
+var ConditionTypes = struct {
+	SiteConfigValidated        ConditionType
+	RenderedTemplates          ConditionType
+	RenderedTemplatesValidated ConditionType
+	Provisioned                ConditionType
+}{
+	SiteConfigValidated:        "SiteConfigValidated",
+	RenderedTemplates:          "RenderedTemplates",
+	RenderedTemplatesValidated: "RenderedTemplatesValidated",
+	Provisioned:                "Provisioned",
+}
```

Comment:
Why are these structs? Why not define them as consts like this:
```go
const (
  SiteConfigValidated ConditionType = "SiteConfigValdated"
  ...
)

Reply:
I can change them to constants - I was merely following the style that we had applied in the TALM and LCA operators.

Reply:
I don't see any reason to use structs.
Additionally it would be possible to edit these at runtime which would be very bad.